components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -169,9 +171,14 @@ private RegistrationResponseProfile createOAuthApplication(RegistrationRequestPr
             //TODO: After implement multi-urls to the oAuth application, we have to change this API call
             if (profile.getRedirectUris().size() == 0) {
                 String errorMessage = "RedirectUris property must have at least one URI value.";
-                throw IdentityException.error(DCRException.class, ErrorCodes.META_DATA_VALIDATION_FAILED.toString(), errorMessage);
+                throw IdentityException.error(DCRException.class, ErrorCodes.META_DATA_VALIDATION_FAILED.toString(),
+                        errorMessage);
+            } else if (profile.getRedirectUris().size() == 1) {
+                oAuthConsumerApp.setCallbackUrl(profile.getRedirectUris().get(0));
```

Comment:
Is there a validation happening for this redirectURI to be a valid URI? If it is not done somewhere else if would be better we do that validation before setting it here.